File: examples/zu_stats.py
# This file contains statistics functions written with Zulu comments
# Uhlelo olulula lokubala izibalo eziyisisekelo kuluhlu lwezinombolo

# Chaza uhlu lwezinombolo zesampula
numbers = [4, 8, 15, 16, 23, 42]

# Bala isamba sazo zonke izinombolo
total = sum(numbers)

average = total / len(numbers)  # Bala uphakathi nendawo ngokuhlukanisa isamba ngokubala kwazo

minimum = min(numbers)  # Thola inani elincane kuluhlu
maximum = max(numbers)  # Thola inani elikhulu kuluhlu

# Shicilela imiphumela kwi-console
print(f"Numbers: {numbers}")
print(f"Sum:     {total}")
print(f"Average: {average:.2f}")
print(f"Min:     {minimum}")
print(f"Max:     {maximum}")

File: examples/zu_stats.rb
# This file contains statistics functions written with Zulu comments

# Chaza uhlu lwezinombolo zesampula
numbers = [4, 8, 15, 16, 23, 42]

# Bala isamba sazo zonke izinombolo
total = numbers.sum

average = total.to_f / numbers.length # Bala uphakathi nendawo

minimum = numbers.min # Thola inani elincane kuluhlu
maximum = numbers.max # Thola inani elikhulu kuluhlu

# Shicilela imiphumela
puts "Izinombolo: #{numbers}"
puts "Isamba:     #{total}"
puts "Uphakathi:  #{format('%.2f', average)}"
puts "Elincane:   #{minimum}"
puts "Elikhulu:   #{maximum}"
